src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclets/formats/html/DeprecatedListWriter.java

```diff
@@ -263,32 +263,34 @@ public DeprecatedListWriter(HtmlConfiguration configuration, DocPath filename) {
      * @throws DocFileIOException if there is a problem writing the deprecated list
      */
     public static void generate(HtmlConfiguration configuration) throws DocFileIOException {
-        DocPath filename = DocPaths.DEPRECATED_LIST;
-        DeprecatedListWriter depr = new DeprecatedListWriter(configuration, filename);
-        depr.generateDeprecatedListFile(
-               new DeprecatedAPIListBuilder(configuration));
+        DeprecatedAPIListBuilder deprAPI = configuration.deprecatedAPIListBuilder;
+        if (deprAPI != null && !deprAPI.isEmpty()) {
```

Comment:
Just a note regarding consistency: everywhere else you are using `configuration.conditionalPages.contains(HtmlConfiguration.PageKind.DEPRECATED)` for this check.

Reply:
Yes, thanks; noted; I guess this is a hangover from before I added the PageKind.